python/lsst/cp/pipe/cpTask.py

```diff
@@ -0,0 +1,594 @@
+        "own pexConfig field.",
+        default=0.05,
+    )
+
+    def setDefaults(self):
+        """Set default config options for the subTasks."""
+        # TODO: Set to proper values
+        self.fe55.temp_set_point = -100
+        self.fe55.temp_set_point_tol = 20
+
+        # TODO: Set to proper values
+        self.readNoise.temp_set_point = -100
+        self.readNoise.temp_set_point_tol = 20
+
+        # TODO: make this work
+        self.brightPixels.temp_set_point = -100
+        self.brightPixels.temp_set_point_tol = 20
+
+        # TODO: find the proper settings for flatPairTask to work. This will mean either using the expTime,
+        # or working out what's wrong with the MONDIODE values (debug that anyway as a large difference might
+        # indicate something else going on). Note that this task doesn't really use much in its config class,
+        # but passes in things at runtime param to its run() method, hence putting in a slot for them here.
+        self.flatPairMaxPdFracDev = 0.99
+
+    def validate(self):
+        """Override of the valiate() method.
+
+        The pexConfigs of the subTasks here cannot be validated in the normal way, as they are the configs
+        for eotest, which does illegal things, and this would require an upstream PR to fix. Therefore, we
+        override the validate() method here, and use it to set the output directory for each of the tasks
+        based on the legal pexConfig parameter for the main task.
+        """
+        log = lsstLog.Log.getLogger("cp.pipe.cpTaskConfig")
+        if not self.eotestOutputPath:
+            raise RuntimeError("Must supply an output path for eotest data. "
+                               "Please set config.eotestOutputPath.")
+
+        taskList = ['fe55', 'brightPixels', 'darkPixels', 'readNoise', 'traps', 'cte', 'flatPair', 'ptc']
+        for task in taskList:
+            if getattr(self, task).output_dir != '.':
+                # Being thorough here: '.' is the eotest default. If this is not the value then the user has
+                # specified something, and we're going to clobber it, so raise a warning. Unlike to happen.
+                log.warn("OVERWRITING: Found a user defined output path of %s for %sTask. "
+                         "This has been overwritten with %s, as individually specified output paths for "
+                         "subTasks are not supported at present"%(getattr(self, task).output_dir,
+                                                                  task, self.eotestOutputPath))
+            getattr(self, task).output_dir = self.eotestOutputPath
+
+
+class CpTask(pipeBase.CmdLineTask):
+    """
+    Calibration Products Production (CP) task.
+
+    This task is used to produce the calibration products required to calibrate cameras.
+    Examples of such operations are as follows:
+    Given a set of flat-field images, find the dark pixels and columns.
```

Comment:
Just because each of those tasks potentially does more than just what it says in the description, and I don't want to be obligated to say every little thing it does.